src/java.base/share/classes/java/lang/AbstractStringBuilder.java

```diff
@@ -67,6 +67,13 @@
      */
     byte coder;
 
+    /**
+     *  The attribute indicates {@code value} is in UTF16 but may be compressible.
```

Comment:
Since we conservatively set this `true` when deleting anything there are state transitions possible where `maybeLatin1` is set to `true` even though the `coder` is still `LATIN1`. Perhaps clarify that such false positives must be handled gracefully? Since a true doesn't imply UTF16 I also suggest a minor rewording like so: `"Indicates that {@code value} might be compressible to LATIN1 if it's UTF16-encoded."`